client/service/apis.go

```diff
@@ -1,130 +1,316 @@
-	return SecretsAPI{Client: c}
+func (c *DatabricksClient) Secrets() SecretsAPI {
+	return SecretsAPI{client: c}
 }
 
 // SecretScopes returns an instance of SecretScopesAPI
-func (c *DBApiClient) SecretScopes() SecretScopesAPI {
-	return SecretScopesAPI{Client: c}
+func (c *DatabricksClient) SecretScopes() SecretScopesAPI {
+	return SecretScopesAPI{client: c}
 }
 
 // SecretAcls returns an instance of SecretAclsAPI
-func (c *DBApiClient) SecretAcls() SecretAclsAPI {
-	return SecretAclsAPI{Client: c}
+func (c *DatabricksClient) SecretAcls() SecretAclsAPI {
+	return SecretAclsAPI{client: c}
 }
 
 // Tokens returns an instance of TokensAPI
-func (c *DBApiClient) Tokens() TokensAPI {
-	return TokensAPI{Client: c}
+func (c *DatabricksClient) Tokens() TokensAPI {
+	return TokensAPI{client: c}
 }
 
 // Users returns an instance of UsersAPI
-func (c *DBApiClient) Users() UsersAPI {
-	return UsersAPI{Client: c}
+func (c *DatabricksClient) Users() UsersAPI {
+	return UsersAPI{client: c}
 }
 
 // Groups returns an instance of GroupsAPI
-func (c *DBApiClient) Groups() GroupsAPI {
-	return GroupsAPI{Client: c}
+func (c *DatabricksClient) Groups() GroupsAPI {
+	return GroupsAPI{client: c}
 }
 
 // Notebooks returns an instance of NotebooksAPI
-func (c *DBApiClient) Notebooks() NotebooksAPI {
-	return NotebooksAPI{Client: c}
+func (c *DatabricksClient) Notebooks() NotebooksAPI {
+	return NotebooksAPI{client: c}
 }
 
 // Jobs returns an instance of JobsAPI
-func (c *DBApiClient) Jobs() JobsAPI {
-	return JobsAPI{Client: c}
+func (c *DatabricksClient) Jobs() JobsAPI {
+	return JobsAPI{client: c}
 }
 
 // DBFS returns an instance of DBFSAPI
-func (c *DBApiClient) DBFS() DBFSAPI {
-	return DBFSAPI{Client: c}
+func (c *DatabricksClient) DBFS() DBFSAPI {
+	return DBFSAPI{client: c}
 }
 
 // Libraries returns an instance of LibrariesAPI
-func (c *DBApiClient) Libraries() LibrariesAPI {
-	return LibrariesAPI{Client: c}
+func (c *DatabricksClient) Libraries() LibrariesAPI {
+	return LibrariesAPI{client: c}
 }
 
 // InstancePools returns an instance of InstancePoolsAPI
-func (c *DBApiClient) InstancePools() InstancePoolsAPI {
-	return InstancePoolsAPI{Client: c}
+func (c *DatabricksClient) InstancePools() InstancePoolsAPI {
+	return InstancePoolsAPI{client: c}
 }
 
 // InstanceProfiles returns an instance of InstanceProfilesAPI
-func (c *DBApiClient) InstanceProfiles() InstanceProfilesAPI {
-	return InstanceProfilesAPI{Client: c}
+func (c *DatabricksClient) InstanceProfiles() InstanceProfilesAPI {
+	return InstanceProfilesAPI{client: c}
 }
 
 // Commands returns an instance of CommandsAPI
-func (c *DBApiClient) Commands() CommandsAPI {
-	return CommandsAPI{Client: c}
+func (c *DatabricksClient) Commands() CommandsAPI {
+	return CommandsAPI{client: c}
+}
+
+// WithCommandMock mocks all command executions for this client
+func (c *DatabricksClient) WithCommandMock(mock CommandMock) {
+	c.commandExecutor = commandExecutorMock{
+		mock: mock,
+	}
+}
```

Comment:
Do we need this here? would someone use this function beyond just testing. Can this be better done with an interface?

Reply:
a) there's nobody using client beyond tf provider itself
b) we need some sort of Dependency Injection to simplify testing of commands. 

this could be implemented more elegantly, I agree.